### PR TITLE
feat(bybit): add pagination cursor 

### PR DESCRIPTION
**Pull request by jorisw**
Currently, I cannot depaginate `fetchMyTrades` against ByBit (`/v5/execution/list`) because `nextPageCursor` is not copied over to the first result set member from the ByBit response.

In this PR, I copy the routine of adding this property, from other functions, to where I can, so that we can depaginate as many endpoints as possible.

### CODE REVIEW

**Comment by carlosmiei** [comment]
Hello @jorisw, thanks that is indeed very important, will review it and merge soon!

File: ts/src/bybit.ts [hunk 1]
```diff
@@ -1148,6 +1148,18 @@ export default class bybit extends Exchange {
         return this.milliseconds () - this.options['timeDifference'];
     }
 
+    addPaginationCursorToResult (response) {
```
Comment (carlosmiei) [inline ts/src/bybit.ts:1151]: @jorisw I noticed we were repeating the same thing repeatedly, so I added this method to DRY it.

**Comment by carlosmiei** [comment]
DEMO

```
p bybit fetchLedger USDT None 1 --sandbox                                                                                                                                                                                   
Python v3.10.9
CCXT v3.1.32
bybit.fetchLedger(USDT,None,1)
[{'account': None,
  'after': 9444.19516595,
  'amount': -2.408e-05,
  'before': 9444.19514187,
  'currency': 'USDT',
  'datetime': '2023-06-10T08:00:00.000Z',
  'direction': 'out',
  'fee': 0.0,
  'id': None,
  'info': {'bonusChange': '0',
           'cashBalance': '9444.19516595',
           'cashFlow': '0',
           'category': 'linear',
           'change': '-0.00002408',
           'currency': 'USDT',
           'fee': '0.00000000',
           'feeRate': '0.0001',
           'funding': '-0.00002408',
           'nextPageCursor': '621%3A1%2C621%3A1',
           'orderId': '1686384000-27-452265-1-2',
           'orderLinkId': '',
           'qty': '1',
           'side': 'Buy',
           'size': '1',
           'symbol': 'ADAUSDT',
           'tradeId': '981192ae-522f-4643-8fd0-d9a660ac5749',
           'tradePrice': '0.2408',
           'transactionTime': '1686384000000',
           'type': 'SETTLEMENT'},
  'referenceAccount': None,
  'referenceId': None,
  'status': None,
  'timestamp': 1686384000000,
  'type': 'trade'}]
```

```
p bybit fetchMyTrades "LTC/USDT" None 1 --sandbox
Python v3.10.9
CCXT v3.1.32
bybit.fetchMyTrades(LTC/USDT,None,1)
[{'amount': 0.05,
  'cost': 5.0,
  'datetime': '2023-04-13T11:36:55.870Z',
  'fee': {'cost': 1.5e-06, 'currency': None},
  'fees': [{'cost': 1.5e-06, 'currency': None}],
  'id': '2120000000005177796',
  'info': {'blockTradeId': '',
           'closedSize': '',
           'execFee': '0.0000015',
           'execId': '2120000000005177796',
           'execPrice': '100',
           'execQty': '0.05',
           'execTime': '1681385815870',
           'execType': 'UNKNOWN',
           'execValue': '5',
           'feeRate': '0.00003',
           'indexPrice': '',
           'isMaker': False,
           'leavesQty': '0',
           'markIv': '',
           'markPrice': '',
           'nextPageCursor': '270%3A0%2C270%3A0',
           'orderId': '1397959978557478912',
           'orderLinkId': '1681385815855678',
           'orderPrice': '999999',
           'orderQty': '0',
           'orderType': 'Market',
           'side': 'Buy',
           'stopOrderType': 'UNKNOWN',
           'symbol': 'LTCUSDT',
           'tradeIv': '',
           'underlyingPrice': ''},
  'order': '1397959978557478912',
  'price': 100.0,
  'side': 'buy',
  'symbol': 'LTC/USDT',
  'takerOrMaker': 'taker',
  'timestamp': 1681385815870,
  'type': 'market'}]
```

```
 p bybit fetchOrders "LTC/USDT:USDT" None 1 --sandbox          
Python v3.10.9
CCXT v3.1.32
bybit.fetchOrders(LTC/USDT:USDT,None,1)
[{'amount': 0.2,
  'average': None,
  'clientOrderId': None,
  'cost': 0.0,
  'datetime': '2023-06-10T10:38:58.577Z',
  'fee': {'cost': 0.0, 'currency': 'USDT'},
  'fees': [{'cost': 0.0, 'currency': 'USDT'}],
  'filled': 0.0,
  'id': 'b8cbef49-dc14-4852-a503-1c83d91bf3b3',
  'info': {'avgPrice': '',
           'blockTradeId': '',
           'cancelType': 'UNKNOWN',
           'closeOnTrigger': False,
           'createdTime': '1686393538577',
           'cumExecFee': '0',
           'cumExecQty': '0',
           'cumExecValue': '0',
           'isLeverage': '',
           'lastPriceOnCreated': '76.97',
           'leavesQty': '0.2',
           'leavesValue': '6',
           'nextPageCursor': 'b8cbef49-dc14-4852-a503-1c83d91bf3b3%3A1686393538577%2Cb8cbef49-dc14-4852-a503-1c83d91bf3b3%3A1686393538577',
           'orderId': 'b8cbef49-dc14-4852-a503-1c83d91bf3b3',
           'orderIv': '',
           'orderLinkId': '',
           'orderStatus': 'New',
           'orderType': 'Limit',
           'placeType': '',
           'positionIdx': '0',
           'price': '30',
           'qty': '0.2',
           'reduceOnly': False,
           'rejectReason': 'EC_NoError',
           'side': 'Buy',
           'slLimitPrice': '0',
           'slTriggerBy': '',
           'smpGroup': '0',
           'smpOrderId': '',
           'smpType': 'None',
           'stopLoss': '',
           'stopOrderType': '',
           'symbol': 'LTCUSDT',
           'takeProfit': '',
           'timeInForce': 'GTC',
           'tpLimitPrice': '0',
           'tpTriggerBy': '',
           'tpslMode': 'UNKNOWN',
           'triggerBy': '',
           'triggerDirection': '0',
           'triggerPrice': '',
           'updatedTime': '1686393538591'},
  'lastTradeTimestamp': 1686393538591,
  'postOnly': False,
  'price': 30.0,
  'reduceOnly': False,
  'remaining': 0.2,
  'side': 'buy',
  'status': 'open',
  'stopLossPrice': None,
  'stopPrice': None,
  'symbol': 'LTC/USDT:USDT',
  'takeProfitPrice': None,
  'timeInForce': 'GTC',
  'timestamp': 1686393538577,
  'trades': [],
  'triggerPrice': None,
  'type': 'limit'}]
```

```
ph bybit fetchLedger null null 1 --sandbox          
PHP v8.2.1
CCXT version :3.1.32
bybit->fetchLedger(, , 1)
Array
(
    [0] => Array
        (
            [id] => 
            [currency] => USDT
            [account] => 
            [referenceAccount] => 
            [referenceId] => 
            [status] => 
            [amount] => -2.408E-5
            [before] => 9444.19514187
            [after] => 9444.19516595
            [fee] => 0
            [direction] => out
            [timestamp] => 1686384000000
            [datetime] => 2023-06-10T08:00:00.000Z
            [type] => trade
            [info] => Array
                (
                    [symbol] => ADAUSDT
                    [side] => Buy
                    [funding] => -0.00002408
                    [orderLinkId] => 
                    [orderId] => 1686384000-27-452265-1-2
                    [fee] => 0.00000000
                    [change] => -0.00002408
                    [cashFlow] => 0
                    [transactionTime] => 1686384000000
                    [type] => SETTLEMENT
                    [feeRate] => 0.0001
                    [bonusChange] => 0
                    [size] => 1
                    [qty] => 1
                    [cashBalance] => 9444.19516595
                    [currency] => USDT
                    [category] => linear
                    [tradePrice] => 0.2408
                    [tradeId] => 981192ae-522f-4643-8fd0-d9a660ac5749
                    [nextPageCursor] => 621%3A1%2C621%3A1
                )

        )

)
```

**Review by carlosmiei — APPROVED** [review]

**Comment by materemias** [comment]
@jorisw thanks for the addition, I am not pretty sure how to resolve my finding hence the necrobumping:
when I call `fetch_ledger()`, the pagination cursor is added to the first result, but in the following line it is filtered out because `parse_ledger()` has no rule for it. In the end it will not be part of the result set returned. This may also be an issue with the other parse functions.

**Comment by carlosmiei** [comment]
> 

@materemias that value is added to the raw response, so you can always get it by checking the `info` object.

Example:
```Python
ledger = await bybit.fetch_ledger()
first = ledger[0]
pagination_cursor = first['info']['nextPageCursor']
```

**Comment by materemias** [comment]
got it, the interesting thing is that because results are returned in timestamp descending order the token is to be found in the last item [-1] (or if multiple trade happened on the same instant like in my case than on the [-2]). I think there is room for improvement here

**Pull request merged**